neo4j/src/test/scala/magnolify/neo4j/ValueTypeSuite.scala

```diff
@@ -76,4 +79,17 @@ class ValueTypeSuite extends MagnolifySuite {
     assert(!fields.map(record.get).exists(_.isNull))
     assert(!record.get("INNERFIELD").get("INNERFIRST").isNull)
   }
+
+  test("AnyVal") {
+    val vt: ValueType[HasValueClass] = implicitly
+    test[HasValueClass]
+
+    val record = vt(HasValueClass(ValueClass("String")))
+    assert(record.get("vc").asString() == "String")
+
+    val v: Value = new StringValue("Hello, world")
+    val a = new MapValue(Map("vc" -> v).asJava)
+    val c = vt.from(a)
+    assert(c == HasValueClass(ValueClass("Hello, world")))
```

Comment:
I think this part is useless (it is already tested in the `test` while doing the round-trip)

Reply:
Not really, since the roundtrip will not test the absence of sub-property in the serialized version. Without the change, it will serialize has `{ "vs": { "str": "Hello, world" }  }` and deserialize to match the original, and the test will pass, even if the case class is an `AnyVal`.

Reply:
What I mean, is that in the line above
```scala
assert(record.get("vc").asString() == "String")
```
We are checking that the value class does not show in the target model.
Since `test` checks the round trip, we know that the conversion back to scala will create the value class in the `HasValueClass`

Reply:
Oh, yeah, the last `assert` may be seen as redundant with `test` + the previous assert. I would be in favor of testing this explicitely, however that's your call.